src/Microsoft.Extensions.Primitives/StringSegment.cs

```diff
@@ -102,23 +191,41 @@ public override int GetHashCode()
             return hash;
```

Comment:
@rynowak How does this allocate? HashCodeCombiner is a struct, is it one of the Add methods that allocates internally?